src/stdlib_string_type.f90

```diff
@@ -0,0 +1,1104 @@
+! SPDX-Identifier: MIT
+
+!> Implementation of a string type to hold an arbitrary sequence of characters.
+!>
+!> This module provides string type compatible with all Fortran instrinsic character
+!> procedures as well as overloaded operators for working with character variables.
+!>
+!> A string type can be easily constructed by creating a new instance from a
+!> character variable or literal by invoking its constructor or by assigning it
+!> to a string type. Generally, the string type behaves similar to a deferred
+!> length character in most regards but adds memory access safety.
+!>
+!> The specification of this module is available [here](../page/specs/stdlib_string_type.html).
+module stdlib_string_type
+    implicit none
+    private
+
+    public :: string_type
+    public :: len, len_trim, trim, index, scan, verify, repeat, adjustr, adjustl
+    public :: lgt, lge, llt, lle, char, ichar, iachar
+    public :: assignment(=)
+    public :: operator(.gt.), operator(.ge.), operator(.lt.), operator(.le.)
+    public :: operator(.eq.), operator(.ne.), operator(//)
+    public :: write(formatted), write(unformatted)
+    public :: read(formatted), read(unformatted)
+
+
+    integer, parameter :: long = selected_int_kind(18)
```

Comment:
Why did you not use one of `iso_fortran_env`?

Reply:
I've been following @sblionel's recommendation to use `selected_*_kind` rather than the `iso_fortran_env` kind parameters so far (see https://stevelionel.com/drfortran/2017/03/27/doctor-fortran-in-it-takes-all-kinds/). This just stuck with me ever since, but I'm can change to use `iso_fortran_env` here as well.

Reply:
I understand the issue (I believe we had a similar discussion when proposing stdlib_kinds). Since it is only used inside this module, it is fine for me as it is.